src/homescreen-progress-header/progress-title.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { getVisibleTasks, ONBOARDING_STORE_NAME } from '@woocommerce/data';
+import { getSetting } from '@woocommerce/settings';
+
+export type ProgressTitleProps = {
+	taskListId: string;
+};
+
+export const ProgressTitle: React.FC< ProgressTitleProps > = ( {
+	taskListId,
+} ) => {
+	const { loading, tasksCount, completedCount, hasVisitedTasks } = useSelect(
+		( select ) => {
+			const taskList = select( ONBOARDING_STORE_NAME ).getTaskList(
+				taskListId
+			);
+			const finishedResolution = select(
+				ONBOARDING_STORE_NAME
+			).hasFinishedResolution( 'getTaskList', [ taskListId ] );
+			const visibleTasks = getVisibleTasks( taskList?.tasks || [] );
+
+			return {
+				loading: ! finishedResolution,
+				tasksCount: visibleTasks?.length,
+				completedCount: visibleTasks?.filter(
+					( task ) => task.isComplete
+				).length,
+				hasVisitedTasks:
+					visibleTasks?.filter(
+						( task ) =>
+							task.isVisited && task.id !== 'store_details'
+					).length > 0,
+			};
+		},
+		[ taskListId ]
+	);
+
+	const title = useMemo( () => {
+		if (
+			! hasVisitedTasks ||
+			completedCount === tasksCount ||
+			completedCount === 0
+		) {
+			const siteTitle = getSetting( 'siteTitle' );
+			return siteTitle
+				? sprintf(
+						/* translators: %s = site title */
+						__( 'Welcome to %s', 'woocommerce' ),
+						siteTitle
+				  )
+				: __( 'Welcome to your store', 'woocommerce' );
+		}
+		switch ( completedCount ) {
+			case 1:
+			case 2:
+				return __(
+					'Only a few more tasks to tick off!',
+					'wc-calypso-bridge'
+				);
+			case 3:
+			case 4:
+			default:
```

Comment:
Was surprised that this didn't take precedence over the bottom case - tested it to make sure it worked. Wonder if it would be better to move the default to the end as is the common practice?

Reply:
Yeah I tested this too (also asked ChatGPT if it was ok 😆 ). I'd agree if you could move it to end, I was torn between the ascending numbers and default positioning 😬